src/openms/include/OpenMS/QC/DBSuitability.h

```diff
@@ -110,33 +142,64 @@ namespace OpenMS
+    * 
+    * Correcting the number of found top deNovo hits with this factor results in them being more comparable to the top
+    * database hits. This in return results in a more linear behaviour of the suitability according to the sampling ratio.
+    * The corrected suitability reflects what sampling ratio your database represents regarding to the theoretical 'perfect'
+    * database. Or in other words: Your database needs to be (1 - corrected suitability) bigger to get a suitability of 1.
+    *
+    * Both the original suitability as well as the corrected one are reported in the result.
     *
     * Since q-values need to be calculated the identifications are taken by copy.
+    * Since decoys need to be calculated for the fasta input those are taken by copy as well.
     *
     * Result is appended to the result member. This allows for multiple usage.
     *
-    * @param pep_ids      vector containing pepIDs with target/decoy annotation coming from a deNovo+database 
-    *                     identification search (currently only Comet-support) without FDR
-    *                     vector is modified internally, and is thus copied
-    * @throws             MissingInformation if no target/decoy annotation is found
-    * @throws             MissingInformation if no xcorr is found
-    * @throws             Precondition if a q-value is found in the input
+    * @param pep_ids            vector containing pepIDs with target/decoy annotation coming from a deNovo+database
+    *                           identification search (currently only Comet-support) without FDR
+    *                           vector is modified internally, and is thus copied
+    * @param exp                MSExperiment that was searched to produce the identifications
+    *                           given in @p pep_ids
+    * @param original_fasta     FASTAEntries of the database used for the ID search (without decoys)
+    * @param novo_fasta         FASTAEntry derived from deNovo peptides
+    * @param search_params      SearchParameters object containing information which adapter
+    *                           was used with which settings for the identification search
+    *                           that resulted in @p pep_ids
+    * @throws                   MissingInformation if no target/decoy annotation is found on @p pep_ids
+    * @throws                   MissingInformation if no xcorr is found,
+    *                           this happends when another adapter than CometAdapter was used
+    * @throws                   Precondition if a q-value is found in @p pep_ids
     */
-    void compute(std::vector<PeptideIdentification> pep_ids);
+    void compute(std::vector<PeptideIdentification> pep_ids, const MSExperiment& exp, std::vector<FASTAFile::FASTAEntry> original_fasta, std::vector<FASTAFile::FASTAEntry> novo_fasta, const ProteinIdentification::SearchParameters& search_params);
```

Comment:
this effectively copies `pep_ids` from the calling scope.
you need to modify the content, right?
the cleanest is probably to use `std::vector<PeptideIdentification>&& pep_ids` and then call it using `compute(std::move(peptides), ....)`...
Or just take by reference, but make it clear in the docs that the content will be modified.

Reply:
also the other vectors could be taken by reference instead of copy?